File: ui-ngx/src/app/shared/models/id/notification-id.ts
// SPDX-FileCopyrightText: Copyright The Thingsboard Authors
// SPDX-License-Identifier: Apache-2.0
import { EntityId } from '@shared/models/id/entity-id';
import { EntityType } from '@shared/models/entity-type.models';

export class NotificationId implements EntityId {
  entityType = EntityType.NOTIFICATION;
  id: string;
  constructor(id: string) {
    this.id = id;
  }
}

File: common/data/src/main/java/org/thingsboard/server/common/data/id/NotificationId.java
// SPDX-FileCopyrightText: Copyright The Thingsboard Authors
// SPDX-License-Identifier: Apache-2.0
package org.thingsboard.server.common.data.id;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import org.thingsboard.server.common.data.EntityType;

import java.util.UUID;

public class NotificationId extends UUIDBased implements EntityId {

    @JsonCreator
    public NotificationId(@JsonProperty("id") UUID id) {
        super(id);
    }

    @Schema(requiredMode = Schema.RequiredMode.REQUIRED, description = "string", example = "NOTIFICATION", allowableValues = "NOTIFICATION")
    @Override
    public EntityType getEntityType() {
        return EntityType.NOTIFICATION;
    }

}
